### ISSUE TITLE
Brave freezes or hangs on launch when there are a large number of `Cookies and other site data`

### ISSUE THREAD

**Issue by bsclifton** [brave/brave-browser#9481]
## Description 
We have several reports from users on all Desktop OSes having a hang or complete freeze of the program when launching.  After investigation, this appears to be due to a large number of entries under brave://settings/content/all

Clearing `Cookies and other site data` will resolve the issue until the user builds up a large number of sites again. @iefremov has characterized the problem (see https://github.com/brave/brave-browser/issues/9481#issuecomment-660111245) and we're working on a fix

## Test plan
1. QA to contact @bsclifton for minimal reproducible profile
2. Put profile into place
3. Launch version of Brave without this fix
4. Confirm it takes a while to load
5. Launch version of Brave with this fix
6. Confirm it loads instantly

## Original issue description
Below is the text from when we were diagnosing this issue. There were multiple problems in play and some of those were split out into separate issues. The text is being kept for posterity

### GitHub issues
**Newly updated Brave freezes after launch - Windows 10 #9372**
    - started happening with `1.7.98` (Chromium 81)
    - multiple users experience freeze for 20 or more seconds, but it's fine after that
    - useful crash dumps shared in issue 🎉 @bsclifton looking into these
    - many users had rewards off or toggled rewards off for troubleshooting

**Brave Beta hangs on start after update #9162**
    - user reported issue with `81.1.8.70` (Chromium 81)
    - UI unusable to the point where menu can't be opened

**System randomly freezing #9294**
    - seems to be GPU related
    - one user reports disabling HW acceleration fixes problem

### Possibly related issues
- Crash on MacOS #9233

### Community issues
- https://community.brave.com/t/newly-updated-brave-1-7-98-chromium-81-0-4044-113-unresponsive-for-3-min-after-launch/120351
- https://community.brave.com/t/open-new-tab-freezes-brave-window-for-2-secs/122291

#### Things that can help us narrow it down
- If you're on macOS, check to see if you have the accessibility keyboard enabled under `System Preferences` > `Accessibility` > `Keyboard` > `Accessibility Keyboard`. There is a known crash if this is enabled... so until this is fixed, it's recommended you disable.
- Try updating/re-installing your video drivers
- Disable Sync unless you're using Brave 1.12 or newer
- Ubuntu users who are using the Snap might try [using the official repo](https://brave-browser.readthedocs.io/en/latest/installing-brave.html)

#### Steps that some people have used to fix the problem
We're working on a fix now - stay tuned.

- Some folks who had toggled rewards and were having hangs every 1 - 3 days where `BAT Ledger` process takes 100% CPU and then goes back down may be fixed as of 1.8 🎉 Please make sure to update to latest version of Brave

*Following these steps will lead to data loss. Please consider backing up your profile before attempting! If you're not sure how to do that, please ask here*

- Some folks have had the problem completely resolved by clicking `Restore settings to their original defaults` on brave://settings/reset to reset their settings. Workaround discovered by @talisto [here](https://github.com/brave/brave-browser/issues/9481#issuecomment-620913686)

- Some folks who had used sync may have hit a condition where duplicate bookmarks were created.  Please see https://github.com/brave/brave-browser/issues/9481#issuecomment-623870404 for more information - kudos to @jonathansampson and a user for narrowing that use-case down 😄 For this case, you can likely try exporting your bookmarks as HTML, clearing your bookmarks in Brave, and then re-importing your bookmarks to see if this helps

### Fixes that have already been released
- `1.8.86` (1.8.x) had a fix for the Brave Rewards publishers list. Instead of being requested by only folks with rewards on, it was being requested by anyone who had ever turned on rewards (even if off). This leads to extra CPU usage when processing.  Captured with https://github.com/brave/brave-browser/issues/9441
- `1.8.86` (1.8.x) had an update for top sites on new tab page which regressed performance upon opening a new tab. Behavior introduced in https://github.com/brave/brave-core/pull/5097 was removed and merged into Nightly (1.10) with https://github.com/brave/brave-core/pull/5490

### Known problems (not being addressed)
- `1.7.98` (1.7 hotfix 1) introduced migration code for cookie and no-script settings. This is only done once, but it could explain some UI hang that folks experience if they have a lot of cookie or no-script settings. @mkarolin looked into this; having over 300 different content settings could cause a problem- but once it finishes, it won't happen again. @bsclifton tried with as many as 100 content settings on an old laptop and it wasn't noticeable. If we wanted to investigate further, we can create an issue and look at https://github.com/brave/brave-core/pull/5453

**Comment by bsclifton** [comment]
For folks that are experiencing lock-ups because of the publisher list being downloaded (for Brave Rewards), there are some great fixes coming in 1.8. One of those fixes a bug where this list continues to be downloaded even after flipping rewards to off
https://github.com/brave/brave-core/pull/5356

**Comment by bsclifton** [comment]
For anyone tracking this who would like to try the 1.8 release candidate, please download and try out https://github.com/brave/brave-browser/releases/tag/v1.8.84 and let us know if this improves behavior 😄

**Comment by riccardoangius** [comment]
Thank you. Clean install is still crashing on my side though.

> **Uploaded Crash Report ID 10280000-64a1-5604-0000-000000000000 (Local Context: 2caabdb7-d67a-4bb5-9da8-4c0c5bd35194)**
> _Crash report captured on Monday, 27 April 2020 at 13:29:56, uploaded on Monday, 27 April 2020 at 13:33:21_

> ╰─$ "/Applications/Brave Browser.app/Contents/MacOS/Brave Browser"        130 ↵
> [771:775:0427/132940.577124:ERROR:browser_switcher_service.cc(238)] XXX Init()
> [771:70147:0427/132941.529922:ERROR:rewards_service_impl.cc(243)] Failed to read file: /Users/ric/Library/Application Support/BraveSoftware/Brave-Browser/Default/rewards_service/confirmations.json
> [781:775:0427/132941.580832:ERROR:confirmations_impl.cc(770)] Failed to load confirmations state, resetting to default values
> [771:26883:0427/132941.997449:ERROR:rewards_service_impl.cc(192)] Failed to read file: /Users/ric/Library/Application Support/BraveSoftware/Brave-Browser/Default/ledger_state
> [781:775:0427/132941.997833:ERROR:ledger_impl.cc(113)] Failed to initialize wallet
> [771:775:0427/132953.873775:ERROR:CONSOLE(0)] "Unchecked runtime.lastError: Cannot access contents of url "chrome://newtab/". Extension manifest must request permission to access this host.", source: chrome-extension://mnojpmjdmbbfmejpflffifhffcmidifd/_generated_background_page.html (0)
> [0427/132956.629002:WARNING:process_memory_mac.cc(93)] mach_vm_read(0x7ffee7e27000, 0x2000): (os/kern) invalid address (1)
> [0427/132956.789760:WARNING:system_snapshot_mac.cc(42)] sysctlbyname kern.nx: No such file or directory (2)
> [0427/132957.004757:WARNING:crash_report_exception_handler.cc(239)] UniversalExceptionRaise: (os/kern) failure (5)
> [813:775:0427/132957.038559:ERROR:child_thread_impl.cc(222)] Invalid PlatformChannel receive right
> [1]    771 illegal hardware instruction  "/Applications/Brave Browser.app/Contents/MacOS/Brave Browser"

**Comment by onmyouji** [comment]
Same here on Windows 10, it still freezes.

[brave.exe_200427_193816.zip](https://github.com/brave/brave-browser/files/4539627/brave.exe_200427_193816.zip)

**Comment by Xyncgas** [comment]
how did the update get to stable by the way

**Comment by bsclifton** [comment]
@riccardoangius for your issue, when was the last time Brave worked for you? Do you know the version you had?

Looking at the call stack, it's a bit hard to tell. But I am seeing some USB keyboard problems. Can you please try turning off your bluetooth keyboard and then launching Brave (mouse-only).

The errors I'm seeing are saying you're missing the entitlement (Catalina thing) for this keyboard to allow bluetooth control

**Comment by bsclifton** [comment]
> how did the update get to stable by the way

This never came up in any of our testing - even now, there are crash reports... but there aren't many clues. Without a solid way to reproduce, it's very difficult to find the root cause ☹️

**Comment by riccardoangius** [comment]
@bsclifton thank you. 

I have no BT keyboard, so I looked around a bit and stumbled [upon this](https://community.brave.com/t/brave-crashes-on-macos-catalina-because-of-accessible-keyboard/105197/3).

Looked around and indeed I hadn't enabled Accessibility Keyboard (as in the linked thread) but *Typing feedback*  was on and **turning it off solved** it all for me.

I was trying the UX for an accessible app I was developing and completely forgot about it 🤦🏻‍♂️

**Comment by bsclifton** [comment]
@riccardoangius this is great! Thanks for finding 😄 I believe we captured the above issue (accessibility keyboard) with https://github.com/brave/brave-browser/issues/7546 but haven't solved it yet

macOS users please do try turning this off, if that is an option for you

**Comment by onmyouji** [comment]
Btw, is there any way to block/disable Brave automatic update?

I tested downgrading to Release Channel v1.7.92 and everything is working fine. But when I went to Help-About and try to update again, the same issue happened.

**Comment by bsclifton** [comment]
@onmyouji yes - I believe as a manual work-around you can install `1.7.92` and then finding the `BraveUpdate.exe`. I believe this is in `c:\Program Files (x86)\BraveSoftware\Brave-Browser`. Use task manager to kill this process and then you can rename this as `BraveUpdate.exe.old` or similar. It shouldn't re-launch

Having it work with 1.7.92 seems like it's a problem with the Chromium 81 upgrade we did. Thanks for helping us troubleshoot this 😄

**Comment by GitOutWest** [comment]
I’ll also mention that [Google] Chrome doesn’t have this problem. So I don't think it's Chrome 81.x all by itself.
Chrome: 81.0.4044.113 (Official Build) (64-bit), and now 81.0.4044.122 (Official Build) (64-bit).

**Comment by Xyncgas** [comment]
I see the freezing on my computer like issue described and after autoupdated

**Comment by intelligentpotato** [comment]
Just checked out the RC2, the issue is still there.

**Comment by the-sushi** [comment]
From #9162
> Thanks for confirming @the-sushi. Do you have to force kill Brave? or does it eventually crash?

I have to force kill it, else it just hangs.

**Comment by bsclifton** [comment]
@intelligentpotato what OS are you on?

As a troubleshooting step, for the more tech savvy folks - you can help us determine if the hanging is a profile-specific problem:
- Profile directory will be under a folder which has a name like `Brave-Browser` / `Brave-Browser-Beta` / `Brave-Browser-Dev` / `Brave-Browser-Nightly`, depending on your release channel.
- on macOS, rename the folder `~/Library/Application Settings/BraveSoftware/<RELEASE_CHANNEL>` out of the way. You can rename to just add `.old` at the end
- on Windows, you can rename the folder at `%userprofile%\AppData\Local\BraveSoftware\<RELEASE_CHANNEL>\User Data`
- on Linux, you can rename the folder at `~/.config/BraveSoftware/<RELEASE_CHANNEL>`

Quit all instances of Brave and then rename the folder. You can relaunch and you'll have a brand new profile. I'm curious if the problem still happens then (try visiting a few websites).  If not, we know the bug is related to something about the user profile

After testing, you can go back to the folder you renamed, remove the new one which got created, and rename the folder back to the old name. 

**If you're not sure what I'm talking about in this post, please don't attempt the above steps 😄**

**Comment by GitOutWest** [comment]
I think several people, including myself, have demonstrated that there's a profile-related issue in play. I had previously tried switching profiles by creating a new one, quitting Brave, then launching into the new one -- no hang. 

I just tried your method, renaming the C:\Users\MyUserID\AppData\Local\BraveSoftware\Brave-Browser\User Data directory and then launching Brave into a new profile. Again, no hang. Previously under my default profile I had manually disabled all extensions (others also used --disable-extensions) but still got the (3-min) hang on launch.

Using your rename\UD method, I just tried with two extension (Acrobat & a password manager), and then quickly tried changing a few settings to set up Brave the way I like it (Download folder, home page, DNT enabled, turn off Brave Rewards button, etc.). After quitting & relaunching Brave, I never got a hang. 

Whatever's going on, it doesn't seem to be a *corrupted* profile but rather something about a specific profile setting (lack of Brave Rewards, no Crypto Wallet, non-use of Tor, etc.).

**Comment by talisto** [comment]
On MacOS 10.14.6, I updated to the Brave 1.8.84 RC but was still getting the lag on startup.  However, I went into Settings > Additional Settings > Reset Settings > "Restore settings to their original defaults", which seems to have resolved it for me.  So it's definitely profile-related.  Using that reset method is much less destructive than torching the whole profile (it keeps bookmarks, history, extension settings, etc), so that's nice!

**Comment by bsclifton** [comment]
@GitOutWest thanks for taking the time to test that 😄 It's definitely appreciated as there are a few different types of issues happening to folks. Can you try the latest 1.8 release candidate? This is very likely what we're going to release soon:
https://github.com/brave/brave-browser/releases/tag/v1.8.86

This 1.8 release does [specifically fix a bug](https://github.com/brave/brave-core/pull/5356) where the publishers list was being pulled even if you have rewards disabled. It also includes some optimizations for when rewards IS enabled and that list is downloaded. Basically, when that list is downloaded, a bunch of records are dropped and inserted in SQLite, which I believe is the freeze that a number of folks are reporting

There's definitely also a hardware related issue- where disabling HW acceleration solves some users. This might be something we see but Chrome doesn't see because Chrome is fetching the field trial values from finch server. We set the flag values at build time and don't reach out- so there's a potential that we have different driver workarounds / exclusion lists than Chrome

All of that said, there is still another two unknown issues: both freezing and crashing is happening to users who have never enabled Rewards... some on a brand new profile. I have several crash dumps and unfortunately, there isn't enough information to narrow the problem down. Having more crash reports and knowing more about the environment (which OS, does it work in Chrome, comparing brave://gpu to chrome://gpu, etc) will help

**Comment by GitOutWest** [comment]
Sorry, I'm going to need a tiny bit of help with that.

When I go to https://github.com/brave/brave-browser/releases/tag/v1.8.86, download the BraveBrowserSetup.exe file (named on my PC BraveBrowserSetup_v1.8.86.exe to make sure I keep things straight) and run it, it does an install. But when I launch Brave from my start menu ("C:\Program Files (x86)\BraveSoftware\Brave-Browser\Application\brave.exe") I'm still seeing Version 1.7.98 Chromium: 81.0.4044.113 (Official Build) (64-bit) in the help, and brave.exe dated 4/19. That's after the 3-min wait until I can do anything.

Is there a different installer I'm supposed to be using from within /v1.8.86? I'm not seeing evidence of another copy of Brave being installed somewhere else (e.g. "C:\Program Files\ or another C:\Program Files (x86)\BraveSoftware tree).

**Comment by onmyouji** [comment]
> @bsclifton Sorry, I'm going to need a tiny bit of help with that.
> 
> When I go to https://github.com/brave/brave-browser/releases/tag/v1.8.86, download the BraveBrowserSetup.exe or BraveBrowserSetup32.exe file (named on my PC BraveBrowserSetup_v1.8.86_RC3.exe, or BraveBrowserSetup32_v1.8.86_RC3.exe to make sure I keep things straight) and run either one, it does an install. But when I launch Brave from my start menu ("C:\Program Files (x86)\BraveSoftware\Brave-Browser\Application\brave.exe") I'm still seeing Version 1.7.98 Chromium: 81.0.4044.113 (Official Build) (64-bit) in the help, and brave.exe dated 4/19. That's after the 3-min wait until I can do anything.
> 
> Is there a different installer I'm supposed to be using from within /v1.8.86? I'm not seeing evidence of another copy of Brave being installed somewhere else (e.g. "C:\Program Files\ or another C:\Program Files (x86)\BraveSoftware tree).

You should download BraveBrowserStandaloneSetup.exe, that's the right one.

**Comment by bsclifton** [comment]
> > @bsclifton Sorry, I'm going to need a tiny bit of help with that.
> > When I go to https://github.com/brave/brave-browser/releases/tag/v1.8.86, download the BraveBrowserSetup.exe or BraveBrowserSetup32.exe file (named on my PC BraveBrowserSetup_v1.8.86_RC3.exe, or BraveBrowserSetup32_v1.8.86_RC3.exe to make sure I keep things straight) and run either one, it does an install. But when I launch Brave from my start menu ("C:\Program Files (x86)\BraveSoftware\Brave-Browser\Application\brave.exe") I'm still seeing Version 1.7.98 Chromium: 81.0.4044.113 (Official Build) (64-bit) in the help, and brave.exe dated 4/19. That's after the 3-min wait until I can do anything.
> > Is there a different installer I'm supposed to be using from within /v1.8.86? I'm not seeing evidence of another copy of Brave being installed somewhere else (e.g. "C:\Program Files\ or another C:\Program Files (x86)\BraveSoftware tree).
> 
> You should download BraveBrowserStandaloneSetup.exe, that's the right one.

This is the right one to download for Windows 😄

**Comment by GitOutWest** [comment]
OK, thanks. That's better. Now showing "Version 1.8.86 Chromium: 81.0.4044.129 (Official Build) (64-bit)" for Brave after I launch it. Unfortunately still with the 3-minute hang :-(

As a side note, before the upgrade (so I was still using 1.7.98 Chromium: 81.0.4044.113 (Official Build) (64-bit) at that time), I ran, one at a time, with each of these directives set in the start menu launch shortcut. And saw no extensions for the appropriate one, so I was doing it right. Still a 3-min freeze before Brave give me control after launch.

"C:\Program Files (x86)\BraveSoftware\Brave-Browser\Application\brave.exe"
  --disable-brave-sync     (to ensure it’s not related to Brave sync)
  --disable-extensions     (to ensure it’s not from an extension)
  --disable-gpu               (in case driver is having a problem)

**Comment by bsclifton** [comment]
@GitOutWest the command should look like this:
```
"C:\Program Files (x86)\BraveSoftware\Brave-Browser\Application\brave.exe" --disable-brave-sync --disable-extensions --disable-gpu
```

Also curious- do you have Rewards enabled?

**Comment by onmyouji** [comment]
Just tried v1.8.86 RC3, it still freezes on launch. No problem if using brand new profile, it only freezes when I'm using my old profile. 

[brave.exe_200429_121845.dmp.zip](https://github.com/brave/brave-browser/files/4549924/brave.exe_200429_121845.dmp.zip)

**Comment by GitOutWest** [comment]
That's what it did look like. I was just trying to save space, but I guess that backfired...

And no, I don't have Rewards enabled.

I tried each of those one at time, with my 3-min freeze of Brave each time on launch. As I noted, I *saw* the missing extensions (e.g. no icons in Brave) in the case of the "--disable-extensions" option. So I'll assume that the other two did their thing also.

"C:\Program Files (x86)\BraveSoftware\Brave-Browser\Application\brave.exe" --disable-brave-sync

"C:\Program Files (x86)\BraveSoftware\Brave-Browser\Application\brave.exe" --disable-extensions

"C:\Program Files (x86)\BraveSoftware\Brave-Browser\Application\brave.exe" --disable-gpu


> @GitOutWest the command should look like this:
> 
> ```
> "C:\Program Files (x86)\BraveSoftware\Brave-Browser\Application\brave.exe" --disable-brave-sync --disable-extensions --disable-gpu
> ```
> 
> Also curious- do you have Rewards enabled?

**Comment by GitOutWest** [comment]
# With the same log output (v1.8.86 RC3 or 1.7.98), if run from a CLI or if added to the Start shortcut and dumped into the chrome_debug.log.

[C:\Program Files (x86)\BraveSoftware\Brave-Browser\Application]brave.exe --args --log-level=2 --enable-logging=stderr

[C:\Program Files (x86)\BraveSoftware\Brave-Browser\Application][19948:4848:0428/224945.376:ERROR:browser_switcher_service.cc(238)] XXX Init()
[28532:28680:0428/224945.389:INFO:confirmations_impl.cc(53)] Initializing confirmations
[28532:28680:0428/224945.390:INFO:confirmations_impl.cc(756)] Loading confirmations state
[28532:28680:0428/224946.698:INFO:confirmations_impl.cc(775)] Successfully loaded confirmations state
[28532:28680:0428/224946.699:INFO:confirmations_impl.cc(736)] Saving confirmations state
[28532:28680:0428/224946.699:INFO:confirmations_impl.cc(736)] Saving confirmations state
[28532:28680:0428/224946.699:INFO:confirmations_impl.cc(736)] Saving confirmations state
[28532:28680:0428/224946.720:INFO:confirmations_impl.cc(752)] Successfully saved confirmations state
[28532:28680:0428/224946.726:INFO:confirmations_impl.cc(752)] Successfully saved confirmations state
[28532:28680:0428/224946.731:INFO:confirmations_impl.cc(752)] Successfully saved confirmations state
[19948:25144:0428/224946.745:ERROR:rewards_service_impl.cc(192)] Failed to read file:

## After 3-min freeze on launch, I can then use Brave. A few more outputs after that below:

[19948:4848:0428/225334.713:ERROR:CONSOLE(1)] "[Shields]: Can't request shields panel data. Error: No tab url specified", source: chrome-extension://mnojpmjdmbbfmejpflffifhffcmidif
d/out/brave_extension_background.bundle.js (1)
[19948:4848:0428/225334.713:ERROR:CONSOLE(1)] "[Shields]: Can't request shields panel data. Error: No tab url specified", source: chrome-extension://mnojpmjdmbbfmejpflffifhffcmidifd/out/brave_extension_background.bundle.js (1)
[19948:4848:0428/225425.731:ERROR:CONSOLE(1)] "[Shields]: Can't request shields panel data. Error: No tab url specified", source: chrome-extension://mnojpmjdmbbfmejpflffifhffcmidifd/out/brave_extension_background.bundle.js (1)
[19948:4848:0428/225425.731:ERROR:CONSOLE(1)] "[Shields]: Can't request shields panel data. Error: No tab url specified", source: chrome-extension://mnojpmjdmbbfmejpflffifhffcmidifd/out/brave_extension_background.bundle.js (1)
[19948:4848:0428/225431.586:ERROR:CONSOLE(1)] "[Shields]: Can't request shields panel data. Error: No tab url specified", source: chrome-extension://mnojpmjdmbbfmejpflffifhffcmidifd/out/brave_extension_background.bundle.js (1)
[19948:4848:0428/225431.586:ERROR:CONSOLE(1)] "[Shields]: Can't request shields panel data. Error: No tab url specified", source: chrome-extension://mnojpmjdmbbfmejpflffifhffcmidifd/out/brave_extension_background.bundle.js (1)

**Comment by bsclifton** [comment]
Thanks for the above @GitOutWest - going to share with some folks. The confirmations state logs are suspicious, especially with rewards off

**Comment by bsclifton** [comment]
@onmyouji can you also try launching via command line with similar args to @GitOutWest to see if you are getting log output mentioning confirmations?

Basically:
```
cd "C:\Program Files (x86)\BraveSoftware\Brave-Browser\Application"
brave.exe --args --log-level=2 --enable-logging=stderr
```

**Comment by onmyouji** [comment]
Yes, I'm also getting confirmation states. I can email you the full logs if you need it. Not sure if it's safe sharing it here because I saw there's payment id there.

![Image 1](https://user-images.githubusercontent.com/8424198/80568068-7ed9a200-8a20-11ea-93e9-d4d28115f0d4.png)

**Comment by bsclifton** [comment]
@onmyouji no need to share - this is great information 😄 I've shared with fellow devs - this is helping narrow down the problem quite a bit

@onmyouji @GitOutWest there is something you both can try...

1. Open the folder `%userprofile%\AppData\Local\BraveSoftware\`
2. Make a copy of the `Brave-Browser` folder and name the copy `Brave-Browser-Nightly`
3. Install and try our Nightly channel at https://brave.com/download-nightly/

Once downloaded installed (it's safe to use both stable Brave and Nightly 😄) you can launch Nightly and it will load the copied profile. There are at least 2 fixes for related issues - I'm curious if these will help

Thanks again for being available to help test and share the result - your help is super appreciated 😄

**Comment by tmancey** [comment]
@onmyouji I am more than happy to jump on a screen share to diagnose the issue, please let me know if this works for you and we can arrange a time, thanks

**Comment by onmyouji** [comment]
@tmancey Thanks, but I prefer to do it through here or the community forum. I'm currently doing something else, don't want to have to be on screen the entire time. 

Should probably check with @GitOutWest if it's okay for him/her.

@bsclifton The Nightly still has the same issue.

![Image 2](https://user-images.githubusercontent.com/8424198/80573953-22c84b00-8a2b-11ea-8aaf-9873d3645509.png)

**Comment by tmancey** [comment]
@onmyouji Sure thanks. I understand you will want to remove your payment id from the log however are there any other lines in the log below what is on the above screenshot as this may help us identify the issue. Thanks

**Comment by GitOutWest** [comment]
Copied Brave-Browser folder => Brave-Browser-Nightly
Installed and running Purple-Brave.
Still see my 3-min freeze on launch.

[C:\Program Files (x86)\BraveSoftware\Brave-Browser-Nightly\Application]brave.exe --args --log-level=2 --enable-logging=stderr

[C:\Program Files (x86)\BraveSoftware\Brave-Browser-Nightly\Application][6704:29500:0429/075142.916:ERROR:browser_switcher_service.cc(238)] XXX Init()
[10780:15004:0429/075142.933:INFO:confirmations_impl.cc(57)] Initializing confirmations
[10780:15004:0429/075142.933:INFO:confirmations_impl.cc(760)] Loading confirmations state
[10780:15004:0429/075144.337:INFO:confirmations_impl.cc(779)] Successfully loaded confirmations state
[10780:15004:0429/075144.337:INFO:confirmations_impl.cc(740)] Saving confirmations state
[10780:15004:0429/075144.338:INFO:confirmations_impl.cc(740)] Saving confirmations state
[10780:15004:0429/075144.338:INFO:confirmations_impl.cc(740)] Saving confirmations state
[10780:15004:0429/075144.366:INFO:confirmations_impl.cc(756)] Successfully saved confirmations state
[10780:15004:0429/075144.374:INFO:confirmations_impl.cc(756)] Successfully saved confirmations state
[10780:15004:0429/075144.381:INFO:confirmations_impl.cc(756)] Successfully saved confirmations state
[6704:5400:0429/075144.389:ERROR:rewards_service_impl.cc(187)] Failed to read file: C:\Users\andrew\AppData\Local\BraveSoftware\Brave-Browser-Nightly\User Data\Default\ledger_state
[10780:15004:0429/075144.400:ERROR:ledger_impl.cc(113)] Failed to initialize wallet
TCC: (Sys) Access is denied.
 "\\.\pipe\chrome.nativeMessaging.out.6d0b69e0a095fc54"
[6704:29500:0429/075600.370:ERROR:CONSOLE(4)] "Polymer is already defined.", source: chrome://settings/crisper.js (4)

------------------------------------
Note: TCC = Take Command Console window. Program launches from that console window via CLI, but doesn't launch from a normal Windows CMD or Windows PowerShell window (normal or admin).

Version 1.10.19 Chromium: 81.0.4044.129 (Official Build) nightly (64-bit)

**Comment by bruce30262** [comment]
Hey guys, would like to post about my case, hope it'll help identifying the issue.  

## ENV
* OS: 64 bit Windows 10, version 1909, build number 18363.778  
* CPU: Intel i7-6700 3.40 GHz
* RAM: 16 GB
* Brave version: 1.8.86 Chromium: 81.0.4044.129 (Official Build) (64-bit)


## Detail
I installed the brave browser with "BraveBrowserStandaloneSetup.exe" ( [v1.8.86](https://github.com/brave/brave-browser/releases/tag/v1.8.86) )  

Then I launched the browser with 

```
brave.exe --args --log-level=2 --enable-logging=stderr
```

Here's the ouput:

```
PS C:\Program Files (x86)\BraveSoftware\Brave-Browser\Application> [20928:19828:0430/012932.226:ERROR:browser_switcher_service.cc(238)] XXX Init()
[1544:21496:0430/012932.262:INFO:confirmations_impl.cc(53)] Initializing confirmations
[1544:21496:0430/012932.262:INFO:confirmations_impl.cc(756)] Loading confirmations state
[1544:21496:0430/012932.626:INFO:confirmations_impl.cc(775)] Successfully loaded confirmations state
[1544:21496:0430/012932.627:INFO:confirmations_impl.cc(736)] Saving confirmations state
[1544:21496:0430/012932.629:INFO:confirmations_impl.cc(736)] Saving confirmations state
[1544:21496:0430/012932.629:INFO:confirmations_impl.cc(736)] Saving confirmations state
[1544:21496:0430/012932.658:INFO:confirmations_impl.cc(752)] Successfully saved confirmations state
[1544:21496:0430/012932.700:INFO:confirmations_impl.cc(752)] Successfully saved confirmations state
[1544:21496:0430/012932.708:INFO:confirmations_impl.cc(752)] Successfully saved confirmations state
[20928:10948:0430/012932.712:ERROR:rewards_service_impl.cc(192)] Failed to read file: C:\Users\Bruce Chen\AppData\Local\BraveSoftware\Brave-Browser\User Data\Default\ledger_state
[1544:21496:0430/012932.716:ERROR:ledger_impl.cc(113)] Failed to initialize wallet
[20928:19828:0430/012944.386:ERROR:os_crypt_win.cc(92)] Failed to decrypt: 參數錯誤。 (0x57)
[20928:19828:0430/012944.386:ERROR:binance_service.cc(305)] Could not decrypt and save Binance token info.
[20928:19828:0430/012946.408:ERROR:device_event_log_impl.cc(162)] [01:29:46.407] Bluetooth: bluetooth_adapter_winrt.cc:1055 Getting Default Adapter failed.
[20928:19828:0430/012958.725:ERROR:CONSOLE(1)] "[object Object]", source: chrome-extension://odbfpeeihdkbihmopkbjmoonfanlbfcl/background.js (1)
[20928:19828:0430/012959.685:ERROR:CONSOLE(1)] "[object Object]", source: chrome-extension://odbfpeeihdkbihmopkbjmoonfanlbfcl/background.js (1)
```
 > "參數錯誤" can be translated to  "Wrong parameters/arguments"

The browser didn't crash though, however for the first few seconds it hangs and the task manager shows that the CPU usage was pretty high. After about 15 seconds everything went back to normal, the browser start working again.

Hope this help !

**Comment by GitOutWest** [comment]
I've never had Brave crash, just hang for a while on launch **. 

On my home system (high end mobo & CPU, very good graphics card, 32G DDR4, DisplayPort monitor, etc.) with v1.8.86 RC3 or 1.7.98, the hang lasts for about 3-min while using 10% of my CPU. 

On my work laptop (older slower model, 16G DDRx, not fancy graphics but using DP) running 1.7.98, the hang lasts about 50-sec while using > 40% of the CPU. So speed (CPU & GPU) doesn't seem to be the deciding factor.

** a couple of times during the hang I've managed to drop the hamburger and launch the Brave Task Manager by "feel" -- all I see is menu outlines for the burger menu then More Tools menu, so I kind of guessed where I needed to hover to launch the TM. I got lucky a couple times, but the TM was just a white window that didn't show me anything. When Brave started responding and filled in the TM window, I *briefly* saw "Browser" sucking a high amount before falling to just a few % (normal). So it didn't *look* like it was the GPU Process or an extension.

**Comment by onmyouji** [comment]
@tmancey Here's the complete log

[chrome_debug.log](https://github.com/brave/brave-browser/files/4555527/chrome_debug.log)


Brave: 1.10.19 Chromium: 81.0.4044.129 (Official Build) nightly (64-bit)
Revision: 3d71af9f5704a40b85806f4d08925db24605ba25-refs/branch-heads/4044@{#979}
OS: Windows 10 OS Version 1809 (Build 17763.1158)

**Comment by the-sushi** [comment]
> on macOS, rename the folder ~/Library/Application Settings/BraveSoftware/<RELEASE_CHANNEL> out of the way. You can rename to just add .old at the end

@bsclifton It's Application Support, not Application Settings, but yes this fixes the hanging, so it does appear to be profile-related.

**Comment by lgeorge** [comment]
Running OS X 10.14.6 (18G4032) on MBP with Brave 1.7.98 Chromium: 81.0.4044.113. 
Brave has been taking about 20 seconds to become responsive after launch. I tried 1.8.84 & still saw a lag. Hadn't tried the CL options yet, but just now restored settings to their defaults per the post from @talisto and it worked. What a relief! Amazing how much difference 20 seconds is when we're used to instant :)

**Comment by bruce30262** [comment]
"Reset Settings" indeed fix the hanging problem, no more high CPU usage while launching the browser ! All I can tell the difference between before/after reset the settings is that the cookies are gone, and the browser extensions are all disabled ( re-enabling those extensions did not hang the browser, everything seems to be fine ). The other settings seems to remain unchanged. 

So the problem might have something to do with the cookies ? I've been using Brave for quite a while ( almost 2 years I guess ) and haven't clear those cookies since the beginning. Maybe it has something to do with those large amount of cookies 🍪

**Comment by onmyouji** [comment]
Yup, can confirm that "Reset settings" fixed the problem. And as @talisto mentioned, you still keep things like bookmarks, history, extension settings, etc. Thanks for sharing.

**Comment by fancycookin** [comment]
Reset Settings also worked for me. Browser no longer freezes on launch.

**Comment by GitOutWest** [comment]
"Reset Settings" also worked for me. Brave now launches very quickly on my "3-min to unfreeze after launch" system. I did have to re-set a few settings per my prefs, enable some extensions, and log back into a bunch of site. I also upgraded my 2nd system (Brave 1.7.98 => 1.8.86), still seeing a 50-sec hang, so I then did the "Reset Settings" and manual re-sets. That system is now OK also.

I don't regard the "Reset Settings" as a real "fix" since it doesn't address the underlying cause that caused Brave to start freezing after a normal upgrade -- "Problems that mysteriously go away mysteriously come back..." Hopefully all the info and logs we've been providing have given the Brave developers some idea of the root cause.

Brave Version 1.8.86 Chromium: 81.0.4044.129 (Official Build) (64-bit)

**Comment by bsclifton** [comment]
> Hopefully all the info and logs we've been providing have given the Brave developers some idea of the root cause.
This is DEFINITELY true - we've identified a few key things and are working on fixes. Thanks everybody for your patience. We don't have a "smoking gun" per-se, but a lot of great things were identified. This wouldn't have been possible without all the good troubleshooting steps shared here by you all 😄

**Comment by Xyncgas** [comment]
it's freezing now whenever opening a tab, instead of opening the browser, can someone shares how to disable updates in brave?

**Comment by Xyncgas** [comment]
I love brave. But the browser is acting up. And I gotta work but not wanting to use chrome but I really wanna control when I can update it or it would just be like those companies that forces every changes and techs that they see should apply to people. it would be no different than the DRM which is your enemy, it would be no different than microsoft forcing update and changing the things in your computer even when it's a bug.

**Comment by Xyncgas** [comment]
I mean I can complain doing it for the people that are not, kinda kidding, I really mean like one of the biggest appeal of the brave browser is to saving time from all those ads right. every ads the browser is blocking and ad block is blocking you see where I am going, it's usually milliseconds right. This laggin like serveral seconds when opening a new tab it's like you hate to see things you enjoy just turning into ( actually never mind I think since I enjoy it I should be patient too ) BTW can you just roll back the public version to before the bug happens already and release the fixed update later.

**Comment by lgeorge** [comment]
+1 @Xyncgas - in general I much prefer to do any updates manually (across all apps & OSs). When there are issues with a series of releases it's great to be able to lag, revert, etc. New feature request? Thanks-

**Comment by eldarmark** [comment]
Reset Settings did not resolve it over here.  New tab window freezes with everything displaying other than the background image and the thumbnails of top sites.

However, deleting all my bookmarks resolved it.  Importing my bookmarks caused the issue to return.  I haven't counted how many I have, but it's a lot, since I don't clean it up at all and import each time I OSRI.  A few thousand, no doubt.

... So why is this new Binance widget parsing my bookmarks?  Is it sending metrics about it to anyone?

**Comment by dilgreen** [comment]
I have experienced this on MacOs repeatedly. I tried a reinstall and the same happened.
![BraveMemoryUsage](https://user-images.githubusercontent.com/5934679/80923489-12420880-8d7c-11ea-9d48-455b7f80948a.png)
Definitely Brave as force-quit immediately frees the system up:
![BraveMemoryUponQuit](https://user-images.githubusercontent.com/5934679/80923483-0bb39100-8d7c-11ea-9764-cb86dbf4fd4f.png)
Once I did this late, and rather than force-quitting, left the machine on, fan qoing full blast. In the morning, Brave was fine, using normal RAM and CPU. It was fine for a week, until I accepted an update - and now it is doing the same again.

**Comment by onmyouji** [comment]
@Xyncgas @eldarmark 
and other users who are experiencing freeze when opening new tab.

Try this extension: https://chrome.google.com/webstore/detail/blank-new-tab-dark-mode/kbgpnmhanjagjnkiekpnkcefiafpapfa

This extension overrides Brave New Tab page with a dark blank page. [Community users mentioned](https://community.brave.com/t/open-new-tab-freezes-brave-window-for-2-secs/122291/10) that this solved their problem.

**Comment by jsecretan** [comment]
@eldarmark do you have Brave Sync enabled?

**Comment by eldarmark** [comment]
@jsecretan  No, I do not have Brave Sync enabled.  I do not believe I have ever had it enabled since my latest OSRI.  In fact... I don't even see Sync in the menu anymore.

I had blamed it on Binance, but thinking about it it could also be:

> Added support for referral background images and top sites on the New Tab Page. (#8218)

@onmyouji Thank you, I will look into that.  I would rather find out why my bookmarks appear to be being parsed by Brave like this, however.

**Comment by jonathansampson** [comment]
I believe many users experiencing these _launch and new-tab_ hangs may have once participate in Brave Sync (now disabled by default), and as a result had potentially been bitten by an aggressive bookmark-duplication bug. This bug would create numerous copies of your bookmarks, causing things to get out of hand quite quickly.

This issue doesn't impact all users with Sync enabled (for instance, I had it enabled at the time of writing this, with no issues whatsoever). To see how many bookmarks you have, and whether you may be in the extreme, check out this [simple extension](https://chrome.google.com/webstore/detail/%C3%BCber-simple-bookmark-coun/mogbbilahcoegijagmcmleheifpijffa/) in the Chrome Web Store.

**Comment by jsecretan** [comment]
@eldarmark would you mind checking Bookmarks.Count.OnProfileLoad from brave://histograms like @jonathansampson posted above to see that is affecting you?

**Comment by eldarmark** [comment]
I've installed the little extension, and the number has currently remained steady at 134 for the past day or so.  This is after I purged my bookmarks file other than my quick bookmarks bar (and discovered that one of the folders therein has some extraneous bookmarks I haven't cleaned out yet).

The Histogram has mention of the original bookmark file:

`Histogram: Bookmarks.Count.OnProfileLoad recorded 1 samples, mean = 6414.0 (flags = 0x61)
0     ... 
5399  ------------------------------------------------------------------------O (1 = 100.0%) {0.0%}
6758  ... 
`

**Comment by jonathansampson** [comment]
Than you for the update, @eldarmark. Now that you've purged your bookmarks, do you still experience the hang associated with new tabs and first-launch?

**Comment by eldarmark** [comment]
As mentioned a few days ago, no, the new tabs thing stopped once I purged my bookmarks in bookmark manager, and returns immediately if I re-import them from the HTML file I exported them to.

Why is Brave parsing my bookmarks on the new tab window load?  Is it taking any telemetry and if so, is it reporting said telemetry to anyone?

**Comment by jonathansampson** [comment]
Excellent question, @eldarmark. This is not for telemetry. What I suspect is happening is the browser begins working to compile a list of potential sites, tabs, and more that you might want to visit. As you type into the address bar, that list is scanned for relevant entries that match your query. Tabs, bookmarks, and more are then suggested. But, if you have quite a few bookmarks, this process could be dramatically slower. Would you be available for a call sometime today? I'd love to chat over Skype or Zoom, and take a closer look at your issue.

**Comment by eldarmark** [comment]
Something about that does not make sense at first glance, as this is new behavior as of 1.8.86, and this bookmark file is extremely old.  To wit, I OSRI'd in September 2019 and the bookmark archive there had ~5200 bookmarks, which I believe I simply re-imported after the OSRI.

Updated to 1.8.90, no change.  Reimported bookmark file, hit 6500~ or so bookmarks, and issue was definitely back.  Clearing out bookmarks seems to resolve it linearly, with 1700 reducing the hiccup to ~1 second or so.

As for Slack / Zoom, sadly, I am not available for that sort of direct phone call / remote desktop troubleshooting / etc.  If there are some debug settings or logging features I can enable that will help, I'd be happy to enable them.

**Comment by jonathansampson** [comment]
Your input is incredibly helpful, @eldarmark. One point of clarification, by OSRI do you mean _Operating System Reinstall_?

**Comment by luciash** [comment]
I get the same issue as @eldarmark since few days ago when Brave was auto-updated. New tab freezes Brave for about 20 seconds. Note: I never used Sync in the past in Brave. But I have lot of bookmarks imported from Chrome too. (So the theory about bookmarks parsing sounds right - the question is why that changed recently? Did have no issues in previous versions of Brave.)

**Comment by luciash** [comment]
Same issue when I just updated right now to Version 1.8.90 Chromium: 81.0.4044.129 (Official Build) (64-bit)

**Comment by jonathansampson** [comment]
@luciash Can you check brave://histograms and share more information regarding your bookmarks? For instance, can you share your `Bookmarks.Count.OnProfileLoad` and `Bookmarks.FileSize`? Those would be very helpful.

**Comment by luciash** [comment]
@jonathansampson 
```
Histogram: Bookmarks.Count.OnProfileLoad recorded 1 samples, mean = 9091.0 (flags = 0x61)
0     ... 
8459  ------------------------------------------------------------------------O (1 = 100.0%) {0.0%}
10589 ... 

Histogram: Bookmarks.FileSize recorded 1 samples, mean = 4960.0 (flags = 0x41)
0     ... 
3202  ------------------------------------------------------------------------O (1 = 100.0%) {0.0%}
5062  ... 
```

**Comment by jonathansampson** [comment]
Thank you, @luciash. It appears you have more than 9,000 bookmarks. Were you expecting to see that many?

**Comment by luciash** [comment]
> Thank you, @luciash. It appears you have 9K+ bookmarks. Were you expecting to see that many?

Yup, kinda expected :) No matter how many bookmarks users have, it should not hang up their browser... (as it did not before)

**Comment by jsecretan** [comment]
Totally agreed @luciash, we'll get to the bottom of it.

**Comment by chico200987** [comment]
Hi guys, like so many users I have the same issue except when I open a new tab, brave is not hanging. It only happens for 20-30s when I open Brave.

I checked the values Bookmarks.Count.OnProfileLoad and Bookmarks.FileSize
`Histogram: Bookmarks.Count.OnProfileLoad recorded 1 samples, mean = 61.0 (flags = 0x61)
0   ... 
61  ------------------------------------------------------------------------O (1 = 100.0%) {0.0%}
76  ... `

`Histogram: Bookmarks.FileSize recorded 1 samples, mean = 24.0 (flags = 0x41)
0   ... 
21  ------------------------------------------------------------------------O (1 = 100.0%) {0.0%}
33  ... `

I'm not sure that 61 bookmarks is too much ?
Windows 10 64 bits
Version 1.8.90 Chromium: 81.0.4044.129 (Official Build) (64-bit)
HW acceleration disabled
3 extensions disabled
Not using Brave Sync

If you want more informations, i'll be happy to try to help you to fix it :)

**Comment by wdeschain** [comment]
> Totally agreed @luciash, we'll get to the bottom of it.

Hi.

I began following this on the 27th of April. My browser is version 1.8.90 Chromium: 81.0.4044.129 (Official Build (64-bit).

My browser somehow corrected itself nearly 36 hours later. I'm a dick for not offering this sooner.

As to why I returned to this thread, I wanted to upgrade a 32gb harddrive to 250gb on a Dell 11 3100 series, and I searched for a solution as to how I could bring my settings and what have you to a reinstall of Brave after the hard drive upgrade.

I found the solution and I found your advice to currently NOT reinstall Brave.

...and that is how I found my way back to this thread.

As I stated previously, on the 27th of April, my browser was absolutely horrible. It would become half ass responsive after about 3-5 minutes of me doing nothing.  I did the suggested all up to the create a new profile.

I did create a new profile, and it functioned as good as new.  The thought of having to customize the browser, however, made me feel ill.  I wanted to load my work profile but because I'm not tech smart, I feared that the old responsive I had escaped would bleed into my new profile and I'd be like screwed from making any more profiles.

Yup. I am tech ignorant 

After some thought and getting some backlog accomplished, I decided to give it a try. I swapped profiles twice(because I had to customized), and lo and behold both profiles functioned as well as ever.

Sorry for the narrative, but it's my craft somewhat, but here is why I am even commenting:

Is there any file or files that I could prepare and then upload them to you folks with the know how to parse the log and maybe find an 'ah hah?'

I've no shame. If there is such a file or folder, plural even, I would hand them over-- privately I should add.

I do all my financials through the browser, and to reaffirm, in the town of Brave, home to a crafting class known as IT professionals, I'm a dummie.

**Comment by KevinL-Grey** [comment]
Was having the problem of brave hanging on startup.  First disabled gpu acceleration and then all extensions.   No go.  Restored default settings.  No go.   Toggled the rewards switch.  That worked. 

 I'm having a separate problem with gpu acceleration so I'm not turning that back on.  Text gets slightly blurry until I scroll.  Then it gets blurry again.    Works fine without gpu acceleration.  The hanging on startup first started when I first turned off GPU acceleration so I thought it was related.

Display device       : GeForce GTX 1080 on GP104-A GPU
Display driver       : 445.87
BIOS                 : 86.04.17.00.a3

**Comment by jonathansampson** [comment]
Hello, @KevinL-Grey. Which version of Brave are you presently using?

**Comment by kobyy** [comment]
this is still happening in Version 1.8.96 Chromium: 81.0.4044.138 (Official Build) (64-bit)
it freezes every time i close Brave and try opening it again

**Comment by dilgreen** [comment]
I see the same behaviour mentioned a few comments back - on launch (but not on opening new tabs or new windows), Brave consumes all available memory and a significant chunk of CPU, fan goes wild, machine becomes unresponsive.
It is actually MacOS kernel helper that seems to be the process concerned, not any Brave process, although it is clearly Brave that is driving this.
As soon as I manage to force quit, everything frees up.
But, if I leave the machine for an hour or so, then Brave functions normally - speedily, memory use is normal - stable for days, until for some reason (update, shutdown or - worst - actually hitting cmd-Q), then I need to go through the whole cycle again.
I saw the speculation that it is the Brave Rewards feature that is causing the problem. I have Brave rewards 'On'.
I need to relaunch to move to the latest update, and will turn rewards 'Off' before doing so, and report back.
But later - I have actual work to do...
MacBook Pro (15-inch, Early 2011) 2.3 GHz Intel Core i7 MacOS 10.13.6 (17G9016)
Brave [Version 1.8.90 Chromium: 81.0.4044.129 (Official Build) (64-bit)](https://brave.com/latest/)

Relaunch
Dil Green
[digital anthropology blog](http://www.digital-anthropology.me)
[Open Credit Network](http://opencredit.network)
@dilgreen on twitter and generally
Use this link to book a half hour slot with me!
https://calendly.com/dilgreen/30-min-video-call

Sent with [ProtonMail](https://protonmail.com) Secure Email.

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Comment by Bachmanity-capital** [comment]
Still happening to me as well

Version 1.8.96 Chromium: 81.0.4044.138 (Official Build) (64-bit)

Opening new tabs is not an issue, only when I'm launching Brave.

**Comment by dilgreen** [comment]
Follow up:
Turned Brave rewards off. Relaunched to install update. Same behaviour - on launch, CPU spike, then gradual ramp up of memory usage over a minute or two until machine is unusable. Leave machine alone, and after some minutes(perhaps less time than before - only 15-20minutes?), everything works perfectly.
Now running: [Version 1.8.96 Chromium: 81.0.4044.138 (Official Build) (64-bit)](https://brave.com/latest/)

Dil Green
[digital anthropology blog](http://www.digital-anthropology.me)
[Open Credit Network](http://opencredit.network)
@dilgreen on twitter and generally
Use this link to book a half hour slot with me!
https://calendly.com/dilgreen/30-min-video-call

Sent with [ProtonMail](https://protonmail.com) Secure Email.

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Comment by wdeschain** [comment]
@dilgreen 

Dunno, but I experienced your exact scenario, I believe.  Two weeks of responses ago in this thread it was suggested to create a new profile, and so I did.

However, I created the profile within the following scenario.
1.  I'm Windows 10 1909.  I had become frustrated with Brave and so to attempt to grind out a little work, I had gone into settings and set my Default Browser to Edge(cut me some slack, I was trying to get caught up on my procrastination).
2.  When I had caught up some, I came back to this thread to see if any solution had been conjured.
3.  It was then I read about creating a new profile.
4.  I left Edge open in the taskbar, and grabbed Brave -- slow, slow, slow, like 3-5 minutes to resolve Google search landing.
5.  I should note that I'm a long time AM Turker, GPT'er, and survey rewards grinder -- this means that my main Brave's Shields only has HTTPS enabled: allow all trackers, allow all cookies, java/scripts allowed, all fingerprinting good(its necessary to have the least resistance with the aforementioned gig).  My Bookmarks Bar is also enabled!  
***this happened to me upon the Binance addition--but not instantly.  That said, both Brave Rewards and Binance were active!
6.  so, Edge is open in the taskbar, set to Default in Windows, I grab Brave, and Brave has the above setup, then hamburger->settings->create new profile AND instantly the New Profile Brave is resolved to  our dashboard. My Home button is enabled, and set to Google...so I test for that landing resolve since before it literally took 3-5 minutes...BUT New Profile Brave snapped clean
7.  I hop over to my "take whatever you want" Brave profile, and it seems fairly normal(its hard to compete with factory profile)
8. I immediately pull up Edge, and X out, parking Brave in the taskbar, go to Windows Settings, and place Brave back as Default
9.  Then, I'm back into Brave, and its as if nothing was ever wrong.
10.  I already had two Profiles, my previously mentioned "friday night thot profile" and my "Sunday morning mass profile," so I didn't need the third.
11. I never even gave my deleting it any consideration, and it did not matter.  My deleting the new profile had no effect on the Brave's performance.  

...sounds super technical, I know.  I am only stating what my complete scenario was when my issues were cleaned up.  I'm a lifelong shade tree-mechanic, but I've no idea how to work under the hood on this vehicle.

Currently, I am: 
Brave
Brave is up to date
Version 1.8.96 Chromium: 81.0.4044.138 (Official Build) (64-bit)

_**SINCE THE LAST UPDATE, MY BRAVE IS SWITCHED ON WITH ACCESS, ALL SHIELDS DOWN,  w/ ONLY UPGRADE HTTPS ON, YET IT IS CALLED OUT BY SITES THAT HAVE AD-BLOCK DETECTION.  I ATTEMPTED TO SEE IF SWITCHING OFF HTTPS UPGRADE WOULD CURE IT.  IT DID NOT.  SO, MY BRAVE SHIELDS HAS EVERTHING SWITCHED OFF, AND ALLOWING ALL THE COOKIES, TRACKERS, AND FINGERPRINTING.  WHEN I AM MET WITH "PLEASE JUST LET ME STICK THE TIP IN BY THE AD-DETECTING SITES," I TURN SHIELD ACCESS OFF AT THE MAIN(THOUGH EVERYTHING IS ALREADY DISABLED)...AND THIS SATISFIES THE SITES AD-DETECTIVE.**_

🖐Lali-Ho🤸‍♀️

**Comment by jonathansampson** [comment]
There is a PR to resolve this startup-hang in version **1.9.x** of Brave. This change has already landed in Brave Beta (brave.com/download-beta) if you would like to check it out. The fix will soon land in our Release build as well.

**Comment by cousinit99** [comment]
> 
> 
> There is a PR to resolve this startup-hang in version **1.9.x** of Brave. This change has already landed in Brave Beta (brave.com/download-beta) if you would like to check it out. The fix will soon land in our Release build as well.

Can I confirm the version you're speaking of, please?  Is that this version?

https://github.com/brave/brave-browser/releases/tag/v1.9.68

**Comment by cousinit99** [comment]
> 
> 
> There is [a PR](https://github.com/brave/brave-core/pull/5490) which resolves this startup-hang in version **1.9.x** of Brave. This change has already landed in Brave Beta (https://brave.com/download-beta) if you would like to check it out. The fix will soon land in our Release build as well.

I just installed Beta 1.10.60, and it still hangs.  This is a fresh install of Brave Beta.  I have never installed it on this machine before, so that rules out anything to do with profiles, bookmarks, user data, etc.

**Comment by jonathansampson** [comment]
@cousinit99 Would you be available for a Zoom call? I'd love to take a closer look at the issue(s) impacting your instance.

**Comment by chico200987** [comment]
I just tested this beta release Version 1.10.60 Chromium: 81.0.4044.138 (Official Build) beta (64-bit)

I have no more hanging issue. I tried to export my bookmarks and then import them, all is fine for me.

**Comment by cousinit99** [comment]
@jonathansampson Yeah, sure.  If it'll help squash this thing.

**Comment by jonathansampson** [comment]
@cousinit99 Send an email to sampson@brave.com at your convenience; I'll respond with a Zoom link immediately and we'll do some cursory profiling of the New Tab Page to see if we can capture some data on this pesky bug. Thank you again for the assistance!

**Comment by cousinit99** [comment]
@jonathansampson Sent.

**Comment by jonathansampson** [comment]
Thank you for the call, @cousinit99. I'm happy that we were able to identify a workaround to your issue (using the `--disable-gpu` flag), and hope to see a more appropriate, long-term solution in place soon.

**Comment by wdeschain** [comment]
@chico200987 

Would you happen to have a link for bookmark/settings export/import build and implementation instruction in Brave?

I'm Windows 10 1909. Brave is wearing Sunday best.

**Comment by wanderleihuttel** [comment]
I've made a test and when I renamed my profile folder Brave starts fast!

**Comment by chico200987** [comment]
> @chico200987
> 
> Would you happen to have a link for bookmark/settings export/import build and implementation instruction in Brave?
> 
> I'm Windows 10 1909. Brave is wearing Sunday best.

Sorry, would yo ulike to know how to export/import bookmarks ? 

If so : 
1 - Ctrl + Shift + O
2 - In the 3 dots menu at the top right "Export bookmarks"
3- Then in the Brave Beta "Import Bookmarks"

That's all :)

**Comment by wdeschain** [comment]
Thank you for responding.

So, I will need Brave Beta version to do this?

I am Brave final version I suppose.

Ctrl, Shift, O opens a Brave explorer, but I have no hamburger top right,
and the actual browser burger is not accessible with the Brave explorer
open.

Does the file have a name?

Your ctrl, shift, O brings me to a manageable wealth of files--I'm unsure
which is the bookmarks.

Regards

**Comment by KevinL-Grey** [comment]
[Version 1.9.72 Chromium: 81.0.4044.138 (Official Build) (64-bit)](https://brave.com/latest/)

Sent with [ProtonMail](https://protonmail.com) Secure Email.

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Comment by Blaster1920** [comment]
Hi all, I'm having the same problem here:
Version 1.9.76 Chromium: 81.0.4044.138 (Official Build) (64-bit)
Every time it starts up it hangs on for about 4+ seconds, nothing responds.
I tried disabling acceleration and extensions but still present.

**Comment by socialnet460** [comment]
Been lurking in this topic for for the past month hoping there will be a fix. I've had the same issue since   the beginning 1.7.x as well. New version update still hasn't solved it including the one that came out today Version 1.9.76 Chromium: 81.0.4044.138 (Official Build) (64-bit)

I timed it and it hangs for me 20 seconds before it works and checking the task manager there's a huge CPU spike (very high power usage) during those 20 seconds.

1) i'm also on windows 10
2) i do not have rewards enabled, although I did to toggle it on and off again since that was one of the suggestions, but that didn't help.
3) I tried disable acceleration and extensions too, didn't help.
4) I tried resetting the browser completely, that helped for a little but the issue returned a few hours later.
5) I don't have a lot of bookmarks either, issue was present even with 0 bookmarks imported
6) on the new tab screen i disabled all the widgets even the background images.

Thanks and hope this is resolve soon.

**Comment by TheFeelTrain** [comment]
I tried moving from 1.9.76 (Stable) to 1.10.76 (Beta) to 1.11.39 (Nightly) and it happens on all three. Hangs for 5-10 seconds before I can actually use anything.

**Comment by kobyy** [comment]
I can confirm  Version 1.9.76 Chromium: 81.0.4044.138 (Official Build) (64-bit) is having the same hang issue
My specs: Rewards turned off, I do have many bookmarks 
I hope Brave isn't turning into Bloatware which I suspect may be causing the issue (trying to load un necesarry data upon load up)

**Comment by luciash** [comment]
Good news! (at least for me) Since yesterday my Brave is not freezing when opening new tab anymore. The tab loads nicely again without freezing the browser. I did change nothing, just got an updated Brave here.
Version 1.9.76 Chromium: 81.0.4044.138 (Official Build) (64-bit)

**Comment by socialnet460** [comment]
> Good news! (at least for me) Since yesterday my Brave is not freezing when opening new tab anymore. The tab loads nicely again without freezing the browser. I did change nothing, just got an updated Brave here.
> Version 1.9.76 Chromium: 81.0.4044.138 (Official Build) (64-bit)

Hi, the freezing issue is mainly on browser launch, not so much on new tab opening. Can you try that? I'm on the newest as well and it still freezes/hangs for 30 seconds on launch.

**Comment by luciash** [comment]
@socialnet460 Okay, might be different issue... but anyway, I have session with many tabs open, so at launch it does not bother me that much as it did when the browser was unusable for 30 seconds on opening new tab previously (as I reported earlier in this thread). I thought it was related problem.

**Comment by socialnet460** [comment]
> @socialnet460 Okay, might be different issue... but anyway, I have session with many tabs open, so at launch it does not bother me that much as it did when the browser was unusable for 30 seconds on opening new tab previously (as I reported earlier in this thread). I thought it was related problem.

Ah I see, okay thanks for the info, I got happy for a little thinking it the other issue was fixed too :)

**Comment by bsclifton** [comment]
@socialnet460 would you be able to jump on a call with us? I'm curious why it hangs. Thanks for providing your info above (https://github.com/brave/brave-browser/issues/9481#issuecomment-636172608 - Windows 10 / Rewards off)

**Comment by socialnet460** [comment]
> @socialnet460 would you be able to jump on a call with us or share your profile (we can provide a secure way to do this)? I'm curious why it hangs. Thanks for providing your info above ([#9481 (comment)](https://github.com/brave/brave-browser/issues/9481#issuecomment-636172608) - Windows 10 / Rewards off)
> 
> When you had tried to reset browser, what did you try exactly? A good way I usually try to reset is to see if the problem is with the profile. You can go to where the profile folder is stored:
> `%USERPROFILE%\AppData\local\BraveSoftware`
> 
> With Brave closed, you can rename the `Brave-Browser` as `Brave-Browser2` for example and then launch to see if the hang is still there. If not, the profile might have an issue

I'm not available for calls, prefer to do it here if possible. The previous method I did to reset was going into settings and clicking Reset Settings. However, I did the way you suggested and renaming it and it no longer hangs which is great, but how certain will this fix things over time I'm not sure but can test over the next few days. With that said, is there way to easy move things from the old profile over or will I just need to redo everything? Thanks!

**Comment by socialnet460** [comment]
> @socialnet460 you can probably go back to your old profile and then export bookmarks- same with passwords too... and then import the information in the new (not slow) profile.
> 
> It would be great if we could get a copy of the slow profile to find the root cause for the hang. Sharing that would entirely be up to you though, as that has your data and sharing understandably is sensitive. We can provide a secure way for you to share if you're able to and trust us 😄 cc: @jonathansampson
> 
> Anyone else reading this too - please consider trying the above (renaming profile) as a debug step. If that fixes, sharing the slow profile with myself or @jonathansampson will enable us to investigate and find/squash the root cause. All copies of profiles will only be installed for debugging purposes and all copies will be deleted after the cause is found

Thanks for your help and sorry I don't feel comfortable sending the profile over, hope you understand! :)

**Comment by jonathansampson** [comment]
@socialnet460 Completely understandable. There are some additional options. For instance, you can create a copy of the profile, and then slowly remove bits and pieces from the copy until there is as little information tied to it as is possible. For example, start by removing Bookmarks. If that resolves the issue, we can investigate the state of your bookmarks (without you needing to share the bookmarks). Next would be removing browsing history and cache. Each step of the way you would check to see if the issue still exists. If the issue still exists, after having removed bookmarks, browsing data, extensions, and more, then the profile is likely in a state that you would be more comfortable sharing for further investigation. Either way, we're here to work with your comfort level, and do what we can to help achieve a better experience for you with Brave 🙂

**Comment by bsclifton** [comment]
I think we've done a good job of diagnosing and breaking out specific issues - and there hasn't been activity in a while.

I'm going to go ahead and close this issue - but please leave comments if you're experiencing and we can re-open. Moving your profile out of the way is a great step to test if the problem is profile related. If so, you can hit me myself or @jonathansampson to help debug further

Huge thanks to everybody that helped diagnose the problem and confirm fixes along the way 😄

**Comment by abishekrsrikaanth** [comment]
I am still having issues with this. I just updated to Version 1.10.97 Chromium: 83.0.4103.116 (Official Build) (64-bit). Anything that I can do to fix this?

**Comment by TheFeelTrain** [comment]
I am still waiting on a fix as well I am staying on Brave 1.5 until it gets fixed because it's completely unbearable for me. I can't stand waiting 30 seconds every time I reopen the browser.

**Comment by Bachmanity-capital** [comment]
I'm also still experiencing these problems. The problem stops when I switch to a new profile, but I can't understand how I import all my passwords, cookies etc. from my old profile. Your guide tells me to select "Old Brave" in the drop-down menu but there is no such option. 

All I've managed to do is transfer bookmarks but nothing else.

**Comment by jcbdmd** [comment]
Registered with GitHub just to comment on this. Longtime Brave user, suffered from linearly increasing length of time of the hang on startup over the last several months, with "Very High" CPU utilization on one thread.

By the time I got fed up enough with it, it was taking **over a minute and a half** to start. That consistent increase over time led me to the idea that it was something that I was doing/adding like rewards, bookmarks, settings, add-ons, etc. that was causing this.

For me it was 100% proven in my testing that it was the JavaScript, Cookies, or Fingerprinting individual "site settings" that was the culprit.

Let me explain... I tighten down my browser to seriously uncomfortable levels and then white-list every site that I would like to have Script access, Cookie access, or Tracking/fingerprinting. Since I setup all 23 of my various workstations the same at both home and work with "white-list only" Brave settings, the computers I browse on most have the longest white-list... It is consistent and repeatable for me that as I white list more and more sites it takes longer and longer for Brave to start once it reaches a certain threshold (maybe 100-150 sites?). The workstation with the longest startup times is my home personal computer, with second being my main work computer in my private office at work. All other workstations, laptops, etc. do not suffer NEAR the same startup slowness.

In all cases the slow startup is instantly fixed by Clear browsing data->Time range->All time and Checking only "cookies and other site data."

That tells me it is either my Script  whitelist or Cookie whitelist as I don't really have too many fingerprinting tracking sites defined.

**Comment by bsclifton** [comment]
This is great information @jcbdmd - thanks for explaining in detail. There is definitely still a problem and while creating a new profile might alleviate the issue, it's a work-around. Re-opened issue so we can continue to look

**Comment by TheFeelTrain** [comment]
Clear browsing data->Time range->All time and Checking only "cookies and other site data." worked for me.

I have around 1,200 bookmarks but that doesn't seem to be the issue at all since clearing the cookies+site data does not clear bookmarks. I had already tried removing all of them and it didn't help so I re-imported them.

**Comment by ohbobva** [comment]
@bsclifton, are you still looking for a profile experiencing the 20-30 second hang at startup?

**Comment by selvidgej** [comment]
I just began experiencing this 3 days ago. The only change I can specifically think of at the time is that I updated by Java environments. 
Clearing Site data also has fixed it for me so far.

**Comment by ohbobva** [comment]
This has been going on, for me, for many months.  It's affecting my work and I'm losing patience.  So, I'm either going to offer my profile directory to help troubleshooting, or switch browsers.

**Comment by angellyna1088** [comment]
> ## Description 
> 
> We have several reports from users on all Desktop OSes having a hang or complete freeze of the program when launching
> 
> 
> 
> ### GitHub issues
> 
> **Newly updated Brave freezes after launch - Windows 10 #9372**
> 
>     - started happening with `1.7.98` (Chromium 81)
> 
>     - multiple users experience freeze for 20 or more seconds, but it's fine after that
> 
>     - useful crash dumps shared in issue 🎉 @bsclifton looking into these
> 
>     - many users had rewards off or toggled rewards off for troubleshooting
> 
> 
> 
> **Brave Beta hangs on start after update #9162**
> 
>     - user reported issue with `81.1.8.70` (Chromium 81)
> 
>     - UI unusable to the point where menu can't be opened
> 
> 
> 
> **System randomly freezing #9294**
> 
>     - seems to be GPU related
> 
>     - one user reports disabling HW acceleration fixes problem
> 
> 
> 
> ### Possibly related issues
> 
> - Crash on MacOS #9233
> 
> 
> 
> ### Community issues
> 
> - https://community.brave.com/t/newly-updated-brave-1-7-98-chromium-81-0-4044-113-unresponsive-for-3-min-after-launch/120351
> 
> - https://community.brave.com/t/open-new-tab-freezes-brave-window-for-2-secs/122291
> 
> 
> 
> ### Recommended user troubleshooting steps
> 
> #### Things that can help us narrow it down
> 
> - If you're on macOS, check to see if you have the accessibility keyboard enabled under `System Preferences` > `Accessibility` > `Keyboard` > `Accessibility Keyboard`. There is a known crash if this is enabled... so until this is fixed, it's recommended you disable.
> 
> - Try updating/re-installing your video drivers
> 
> - Launch with command line arguments
> 
>     - `--disable-brave-sync` (to ensure it's not related to Brave sync)
> 
>     - `--disable-extensions` (to ensure it's not from an extension)
> 
>     - `--disable-gpu` (in case driver is having a problem)
> 
> - Ubuntu users who are using the Snap might try [using the official repo](https://brave-browser.readthedocs.io/en/latest/installing-brave.html)
> 
> 
> 
> #### Steps that some people have used to fix the problem
> 
> We're still diagnosing the problems and working on real fixes. For folks who are open to experimentation, there are some known work-arounds.
> 
> 
> 
> - Some folks who had toggled rewards and were having hangs every 1 - 3 days where `BAT Ledger` process takes 100% CPU and then goes back down may be fixed as of 1.8 🎉 Please make sure to update to latest version of Brave
> 
> 
> 
> *Following these steps will lead to data loss. Please consider backing up your profile before attempting! If you're not sure how to do that, please ask here*
> 
> 
> 
> - Some folks have had the problem completely resolved by clicking `Restore settings to their original defaults` on brave://settings/reset to reset their settings. Workaround discovered by @talisto [here](https://github.com/brave/brave-browser/issues/9481#issuecomment-620913686)
> 
> 
> 
> - Some folks who had used sync may have hit a condition where duplicate bookmarks were created.  Please see https://github.com/brave/brave-browser/issues/9481#issuecomment-623870404 for more information - kudos to @jonathansampson and a user for narrowing that use-case down 😄 For this case, you can likely try exporting your bookmarks as HTML, clearing your bookmarks in Brave, and then re-importing your bookmarks to see if this helps
> 
> 
> 
> ### Areas that Brave team members are looking into (issues to be created)
> 
> - Sync has had issues with duplicate bookmarks. @jonathansampson has been working with users to understand and characterize this problem. 
> 
> 
> 
> ### Fixes that have already been released
> 
> - `1.8.86` (1.8.x) had a fix for the Brave Rewards publishers list. Instead of being requested by only folks with rewards on, it was being requested by anyone who had ever turned on rewards (even if off). This leads to extra CPU usage when processing.  Captured with https://github.com/brave/brave-browser/issues/9441
> 
> - `1.8.86` (1.8.x) had an update for top sites on new tab page which regressed performance upon opening a new tab. Behavior introduced in https://github.com/brave/brave-core/pull/5097 was removed and merged into Nightly (1.10) with https://github.com/brave/brave-core/pull/5490
> 
> 
> 
> ### Known problems (not being addressed)
> 
> - `1.7.98` (1.7 hotfix 1) introduced migration code for cookie and no-script settings. This is only done once, but it could explain some UI hang that folks experience if they have a lot of cookie or no-script settings. @mkarolin looked into this; having over 300 different content settings could cause a problem- but once it finishes, it won't happen again. @bsclifton tried with as many as 100 content settings on an old laptop and it wasn't noticeable. If we wanted to investigate further, we can create an issue and look at https://github.com/brave/brave-core/pull/5453

**Comment by angellyna1088** [comment]
open

**Comment by bsclifton** [comment]
@ohbobva yes - absolutely 😄 You can try the following to minimize your impact:
- backup your profile directory (make a zip, etc). On macOS, it's at `~/Library/Application Support/BraveSoftware/Brave-Browser`; on Windows it's at `%USERPROFILE%\AppData\Local\BraveSoftware\Brave-Browser`
- now that your profile is backed up, you can delete anything which is considered sensitive (bookmarks, saved passwords, etc)
- once you're comfortable with profile, zip this one up and send over to me at `clifton@brave.com`

I promise to treat this data as if it were my own. No copies will be made and when done troubleshooting it will be deleted. The data will be purely used to reproduce the problem

**Comment by jschiarizzi** [comment]
> Clear browsing data->Time range->All time and Checking only "cookies and other site data." worked for me.
> 
> I have around 1,200 bookmarks but that doesn't seem to be the issue at all since clearing the cookies+site data does not clear bookmarks. I had already tried removing all of them and it didn't help so I re-imported them.

This worked for me on Windows 10 with latest brave.

**Comment by ohbobva** [comment]
> @ohbobva yes - absolutely 😄 You can try the following to minimize your impact:
> 
> * backup your profile directory (make a zip, etc). On macOS, it's at `~/Library/Application Support/BraveSoftware/Brave-Browser`; on Windows it's at `%USERPROFILE%\AppData\Local\BraveSoftware\Brave-Browser`
> * now that your profile is backed up, you can delete anything which is considered sensitive (bookmarks, saved passwords, etc)
> * once you're comfortable with profile, zip this one up and send over to me at `clifton@brave.com`
> 
> I promise to treat this data as if it were my own. No copies will be made and when done troubleshooting it will be deleted. The data will be purely used to reproduce the problem

Email sent.

**Comment by cjwijtmans** [comment]
Something is noticed is that one of my cores ramps up to 100% and then it stops and another core ramps up to 100% then when it stops doing that the stalling is over and the browser is started. What is happening here? Secondly there is a design flaw somewhere that needs to be put in a worker thread that wont block the main application thread. If that bug persists at least it wont block the main thread.

**Comment by bsclifton** [comment]
Thanks @ohbobva - I'm able to reproduce and it definitely does seem tied to the cookies / site data setting. Will continue debugging

**Comment by iefremov** [comment]
Many thanks @ohbobva , we finally were able to nail down the problem, will fix it soon.
Some details:
 1. To quickly get a startup performance profile, one can launch the browser with `--trace-startup --trace-startup-file=./trace.json --trace-startup-duration=30` and then open the trace using `brave://tracing`
 2. Something strange is defintely happening at `OnPolicyChanged` - dozens of calls, ~130ms each 
<img width="1254" alt="Screenshot 2020-07-17 at 19 56 39" src="https://user-images.githubusercontent.com/3109556/87789421-3a181700-c869-11ea-8414-04ec617cab26.png">
 
3. This is caused by the following chain of events:
`BravePrefProvider::UpdateCookieRules` -> `BravePrefProvider::NotifyChanges` -> `PrefProvider::Notify` -> `CookieSettings::OnContentSettingChanged` -> `ExtensionSpecialStoragePolicy::OnCookieSettingChanged` -> `ExtensionSpecialStoragePolicy::NotifyPolicyChangedImpl` -> `SpecialStoragePolicy::NotifyPolicyChanged` ->  `DOMStorageContextWrapper::OnPolicyChanged` -> `DOMStorageContextWrapper::OnStoragePolicyChanged`.

 4. Now `DOMStorageContextWrapper::OnStoragePolicyChanged` iterates over all origins stored in Local State (~1500 ones) - this takes 140 ms (on a powerful machine) on the UI thread.

 5. This notification is triggered as many times as there are rule updates produced by `BravePrefProvider::UpdateCookieRules`. In the specific case, there are several dozens of shields down exceptions, each triggers a loop over Local Storage Domains, this is what throws a spanner in the works.

cc @bsclifton @bridiver

**Comment by TheFeelTrain** [comment]
Very cool. I'm glad we were finally able to get to the bottom of this. Thank you for the ongoing support & development!

**Comment by bsclifton** [comment]
Updated to a P2 now that we've been able to find the cause and we know the impact. Definitely something we would consider hotfixing for

**Comment by bsclifton** [comment]
Assigning @bridiver as he had a possible fix

**Comment by bsclifton** [comment]
Quick update for everybody watching this - we have a fix and are preparing a hotfix release 😄👍 

Huge thanks to:
- @ohbobva for providing a profile that reproduces the problem
- @iefremov for diagnosing the issue to find the root cause
- @bridiver for providing a root cause fix 😄

**Comment by LaurenWags** [comment]
Verified passed with

```
Brave | 1.11.101 Chromium: 84.0.4147.89 (Official Build) (64-bit)
-- | --
Revision | 19abfe7bcba9318a0b2a6bc6634a67fc834aa592-refs/branch-heads/4147@{#852}
OS | macOS Version 10.14.6 (Build 18G3020)
```

* Reproduced the hang on launch using `1.11.97` and minimal profile. Launched 5x and on my machine observed an average of ~15s hang each time.
* Launched profile with `1.11.101` and launched 5x. On my machine I observed this went down to an average of ~3s.
* Confirmed able to set cookies/site data without issue.

-----
